internal/handlers/common/update.go

```diff
@@ -340,12 +340,8 @@ func processIncFieldExpression(doc *types.Document, updateV any) (bool, error) {
 
 		incremented, err := addNumbers(incValue, docValue)
 		if err == nil {
-			err := doc.SetByPath(path, incremented)
-			if err != nil {
-				return false, NewWriteErrorMsg(
-					ErrUnsuitableValueType,
-					fmt.Sprintf(`Cannot create field in element {%s: %v}`, path.Prefix(), docValue),
-				)
+			if err = doc.SetByPath(path, incremented); err != nil {
+				return false, lazyerrors.Error(err)
```

Comment:
There is no test case for it, and I couldn't come up with one quickly, so I return `lazyerrors.Error` to make the fact we don't test it explicit.